src/data_info/services/entry.clj

```diff
@@ -38,19 +40,20 @@
 ;; file specific
 
 (defn- get-file
-  [irods user path]
-    (if (zero? (item/file-size irods path))
-      (clean-return irods "")
-      (proxy-input-stream-return irods (ops/input-stream irods path))))
+  [irods user path istream-ref]
+  (if (zero? @(rods/file-size irods user (cfg/irods-zone) path))
+    (clean-return (:jargon irods) "")
+    (proxy-input-stream-return (:jargon irods) @istream-ref)))
 
 (defn- file-entry
-  [cm path {:keys [user attachment]}]
+  [irods path {:keys [user attachment]}]
   (let [filename    (str \" (file/basename path) \")
+        istream-ref (delay (io/input-stream (ops/input-stream @(:jargon irods) path)))
```

Comment:
This `io/input-stream` is what makes the input stream reusable, and thus lets Tika use it once and then have us still return it from the start below.